Arch Linux support

I wanted to give intentguard a try, but noticed it failing with:

```
test_foo.py:1: in <module>
    import intentguard as ig
../micromamba/envs/tmp/lib/python3.13/site-packages/intentguard/__init__.py:10: in <module>
    llamafile = Llamafile()
../micromamba/envs/tmp/lib/python3.13/site-packages/intentguard/infrastructure/llamafile.py:165: in __init__
    raise Exception(f"llamafile exited with status {status}")
E   Exception: llamafile exited with status 1
```

I'm on linux, so I suspect the problem stems from the llamafile being an `.exe` file.

The file was present (downloaded on the fly).

System info:
- Python 3.13.0
- OS: Arch Linux

---

Happy to contribute if this is desired 🙂

--- a/intentguard/infrastructure/llamafile.py
+++ b/intentguard/infrastructure/llamafile.py
@@ -130,16 +130,13 @@ class Llamafile(InferenceProvider):
         ]
 
         system = platform.system()
-        if system == "Darwin":  # macOS
-            # Make llamafile executable on macOS
+        if system != "Windows":
+            # On non-Windows, make llamafile executable, and run in sh
             try:
                 os.chmod(str(llamafile_path), 0o755)  # rwxr-xr-x
                 logger.debug(f"Made llamafile executable at {llamafile_path}")
             except OSError as e:
                 logger.warning(f"Failed to make llamafile executable: {e}")
-
-        if system != "Windows":
-            # On non-Windows, run in sh
             command.insert(0, "sh")
 
         self.process = subprocess.Popen(

